cmd/shipperctl/cmd/chart/command.go

```diff
@@ -0,0 +1,142 @@
+package chart
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/bookingcom/shipper/cmd/shipperctl/kubeconfig"
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	shipperchart "github.com/bookingcom/shipper/pkg/chart"
+	"github.com/bookingcom/shipper/pkg/chart/repo"
+	"github.com/bookingcom/shipper/pkg/client"
+)
+
+var Command = &cobra.Command{
+	Use:   "chart",
+	Short: "operate on Shipper Charts",
+}
+
+const (
+	AgentName = "shipperctl/chart"
+)
+
+var (
+	namespace      string
+	appName        string
+	releaseName    string
+	kubeConfigFile string
+)
+
+func init() {
+	renderCmd := &cobra.Command{
+		Use:   "render --app bikerental",
+		Short: "render Shipper Charts for an Application or Release",
+		RunE:  renderChart,
+	}
+
+	kubeconfig.RegisterFlag(renderCmd.Flags(), &kubeConfigFile)
+	renderCmd.Flags().StringVarP(&namespace, "namespace", "n", "", "namespace where the app lives")
+	renderCmd.Flags().StringVar(&appName, "app", "", "application that will have its chart rendered")
+	renderCmd.Flags().StringVar(&releaseName, "release", "", "release that will have its chart rendered")
+
+	Command.AddCommand(renderCmd)
+}
+
+func renderChart(cmd *cobra.Command, args []string) error {
+	c, err := newChartRenderConfig()
+	if err != nil {
+		return err
+	}
+
+	chartFetcher := newFetchChartFunc()
+	chart, err := chartFetcher(&c.ChartSpec)
+	if err != nil {
+		return err
+	}
+
+	rendered, err := shipperchart.Render(
+		chart,
+		c.ReleaseName,
+		c.Namespace,
+		c.ChartValues,
+	)
+	if err != nil {
+		return err
+	}
+
+	for _, v := range rendered {
+		fmt.Printf("%s\n---\n", v)
```

Comment:
Consider printing to a buffer and returning a rendered string rather than a direct flush to STDOUT. The control flow will gain more ownership over the rendered data.

Reply:
I'd agree if we eventually move parts of this to a library kind of package, but `renderChart` here is the top level func executed by the command, it's got nowhere to return data to.